### ISSUE TITLE
JSHint return an invalid response, check your console for more info

### ISSUE THREAD

**Issue by roqueorts** [AtomLinter/linter-jshint#258]
I get this error when I open one specific file.

**Comment by hasanlussa** [comment]
+1

**Comment by gregtyler** [comment]
I'm seeing this too, since upgrading to Atom 1.7.0 this morning, in all JavaScript files I open.

I'm running version 1.11.4 of linter and 2.1.0 of linter-jshint. My operating system is Windows 7.

Opening the console, the error is "SyntaxError: Unexpected end of input". Trace:
```
(anonymous function)	@	main.js:126
step	@	/C:/Users/ggiltrow/.atom/packages/linter-jshint/lib/main.js:3
_tickCallback	@	node.js:392
```

**Comment by icaruus** [comment]
Same for me, Atom 1.7.0, Windows 7, linter-jshint 2.1.0.
Console:

`[Linter-JSHint] SyntaxError: Unexpected end of input(…)
(anonymous function) @ main.js:126`

**Comment by BlaM** [comment]
Same here, exactly stating at the moment Atom updated to 1.7.0. However the error is reported in line 123, not in 126 for me:

    [Linter-JSHint] SyntaxError: Unexpected end of input
    at Object.parse (native)
    at Object.<anonymous> (.../packages/linter-jshint/lib/main.js:124:25)
    at [object Generator].next (native)
    at step (...\packages\linter-jshint\lib\main.js:3:273)
    at process._tickCallback (node.js:392:9)

      (anonymous function) @ .../packages/linter-jshint/lib/main.js:123

Windows 8.1 for me

**Comment by gregtyler** [comment]
I've done a bit of loose digging, and it looks like the `Helpers.execNode` command returns an empty string rather than JSON as expected. Running the same command in the terminal though, I get the expected response back. So the command is fine, but running it from Node isn't working.

I believe `Helpers.execNode` traces back to the "exec" function in `node_modules/atom-linter/lib/helpers.js`, but I'm not particularly familiar with node internals and yields, so I can't really dig much further.

**Comment by Whipgit** [comment]
+1

**Comment by cakenggt** [comment]
+1
```[Linter-JSHint] SyntaxError: Unexpected end of input(…) (anonymous function) @ /C:/Users/cakenggt/.atom/packages/linter-jshint/lib/main.js:123```

**Comment by ZStelmo** [comment]
Same where.

**Comment by champunes** [comment]
+1
I detected the problem after auto-update of my Atom to 1.7.0 in Windows.

**Comment by dhprototype** [comment]
Exact same error as @Whipgit 

Windows 7 Pro
Atom 1.7.0
linter-jshint: 2.1.0
atom-linter: 4.6.1

**Comment by cristian-sima** [comment]
+1

**Comment by Arcanemagus** [comment]
Closing as a duplicate of #238, it definitely looks like something broke in Atom's BufferedNodeProcess (which `execNode` wraps around) with the v1.7.0 update on Windows.
